### ISSUE TITLE
Error with "rc" loss. 

### ISSUE THREAD

**Issue by eminSerin** [ngohgia/brain-surf-cnn#3]
Hi Gia, 

Firstly, I want to thank you for developing the very interesting model and for the paper (i.e., Ngo et al., 2022). I am trying to reproduce the results shown on the paper following the instructions given in the repository. However, finetuning (i.e., 5_finetune_with_rc_loss.sh) throws an assert error, which originates in two following codes: 

https://github.com/ngohgia/brain-surf-cnn/blob/4d2c9fc661e6ec50cbe6e699ea02c0e66d41b14f/utils/experiment.py#L69

https://github.com/ngohgia/brain-surf-cnn/blob/4d2c9fc661e6ec50cbe6e699ea02c0e66d41b14f/utils/utilities.py#L72

I think the problem is that, the "contrast_mse_loss" function requires the output shape to be (2, 94, 32492), however "experiment.eval()" calls the loss function within the mini-batch loop, thereby providing (1, 94, 32492). Do you think it is a minor bug in the loss function, or am I doing something wrong? 

Thanks a lot for your help!

**Comment by ngohgia** [comment]
Hi Emin, thanks for your interest and kind words!
I think your guess on the bug might be correct. Can you try having `val_loader` with bach size of 2 instead?

**Comment by eminSerin** [comment]
Hi Gia, 
Thanks for your quick response. I did not change any hyperparameters, so `val_loader ` is already set to batch size of 2 as default. I think the thing is that the number of validation subjects is set to 5 in the experiments scripts you provided (see below), meaning, given `batch_size=2`, only one subject will be fed into the model in the third iteration of the mini-batch.  Are the parameters written in each .sh file the same parameters used to produce the results shown in Ngo et al., 2022? If not, please give me the list of subjects and the parameters. 

https://github.com/ngohgia/brain-surf-cnn/blob/4d2c9fc661e6ec50cbe6e699ea02c0e66d41b14f/experiments/MICCAI2020/scripts/5_finetune_with_rc_loss.sh#L10

**Comment by ngohgia** [comment]
The subject list is here https://github.com/ngohgia/brain-surf-cnn/blob/4d2c9fc661e6ec50cbe6e699ea02c0e66d41b14f/data/MICCAI2020/HCP_train_val_subj_ids.csv

- Are you re-running the experiments on the HCP dataset?
- Besides the NUM_VAL_SUBJ, i.e. number of subjects, there is another hyperparameter which is NUM_SAMPLES = 8, did you also generate 8 samples from each subject's resting-state timeseries?
- Is the assertion error raised for every iteration or as you pointed out, only for the 3rd iteration of the mini-batch?

**Comment by eminSerin** [comment]
- Yes, I am using the HCP dataset. 
- What's the main purpose of using 8 samples from each rs timeseries?
- No only the 3rd iteration because there is only one validation subject left in this iteration when NUM_VAL_SUBJ=5.
